### ISSUE TITLE
Non-snapshot build fails in Travis due to outdated gem flags

### ISSUE THREAD

**Issue by headius** [jruby/jruby#6103]
This appears to be failing only in non-snapshot builds. It fails at least on Travis, but does not appear to prevent the actual local release process.

```
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ jruby-tests ---
[INFO] 
[INFO] --- maven-enforcer-plugin:1.0:enforce (default) @ jruby-tests ---
[INFO] 
[INFO] --- maven-enforcer-plugin:1.0:enforce (enforce-maven) @ jruby-tests ---
[INFO] 
[INFO] --- gem-maven-plugin:1.0.10:initialize (default) @ jruby-tests ---
[INFO] ERROR:  While executing gem ... (OptionParser::InvalidOption)
[INFO]     invalid option: --no-rdoc
```

JRuby 9.2.10 updated to RubyGems 3.0.6, which has replaced the `--no-rdoc` and --no-ri` flags with a single `--no-document` flag. I believe the gem-maven-plugin needs to be updated to use the newer flag, if it is indeed using the JRuby version from the project basedir.

An example failed build is here: https://travis-ci.org/jruby/jruby/builds/657473859

cc @mkristian

**Comment by olleolleolle** [comment]
Thinking out loud.

Would it be enough to create a `--document ''` option, which would pick "no formats" in:

-  [gem-maven-plugin/src/main/java/de/saumya/mojo/gem/GemifyMojo.java](https://github.com/torquebox/jruby-maven-plugins/blob/d835086c66e9f57d274047cd0505a278d8d54c36/gem-maven-plugin/src/main/java/de/saumya/mojo/gem/GemifyMojo.java#L174) 
-  [gem-maven-plugin/src/main/java/de/saumya/mojo/gem/InstallMojo.java](https://github.com/torquebox/jruby-maven-plugins/blob/d835086c66e9f57d274047cd0505a278d8d54c36/gem-maven-plugin/src/main/java/de/saumya/mojo/gem/InstallMojo.java#L72) 

Existing gem options could remain (`installRDoc`, `installRI`), but be used to output a `--document 'rdoc,ri'`-like string.

**Comment by headius** [comment]
@olleolleolle Yeah that might do it. It occurs to me now I'm not sure why an older JRuby is getting hooked up to a newer RubyGems in the first place; I'd have expected it to use the one it shipped with. So it might be possible to pin it to an older version, though updating it to latest would obviously be best.

**Comment by headius** [comment]
This is still a problem and tripped me up building on the jruby-9.2 branch. The VERSION file had not been updated to 9.2.12, resulting in us attempting to build a release version. The problem lies in this configuration: (this is from my 9.2 backport PR):

https://github.com/jruby/jruby/blob/82bedae06cea4b3330661383b9fe1fbec0e8ca71/test/pom.rb#L63-L68

Releasing a new plugin that doesn't use these deprecated flags is one option. The other is to just use JRuby 9.2.7.0 on both paths, which is what I'll do to patch it for now.

**Comment by headius** [comment]
Ok, forcing it to a specific version and building on the snapshot leads to another problem: it attempts to install jruby-launcher using the 9.2.7.0 jar and the current jruby.home, leading to this error from

```
[INFO] LoadError: no such file to load -- rbconfig
[INFO]   require at org/jruby/RubyKernel.java:974
[INFO]    <main> at /Users/headius/projects/jruby/lib/ruby/gems/shared/gems/jruby-launcher-1.1.6-java/siteconf20200617-83059-5knf5f.rb:1
[INFO]   require at org/jruby/RubyKernel.java:974
```

I'm digging deeper.

**Comment by headius** [comment]
The PR I mention above, for backports to 9.2.12, is #6285.

**Comment by headius** [comment]
Ok, the rbconfig issue is also a special big of logic only run when non-SNAPSHOT:

https://github.com/jruby/jruby/blob/82bedae06cea4b3330661383b9fe1fbec0e8ca71/test/pom.rb#L120

So, the rdoc flags would normally prevent getting to this point, and if that's fixed, then the improper jruby home causes this to fail.
